frontend/components/PaginatedList/PaginatedList.tsx

```diff
@@ -206,71 +215,80 @@ function PaginatedListInner<TItem extends Record<string, any>>(
           <Spinner />
         </div>
       )}
-      <div>
-        <ul className={`${baseClass}__list`}>
-          {heading && (
-            <li className={`${baseClass}__row ${baseClass}__header`}>
-              {heading}
+      <ul className={`${baseClass}__list`}>
+        {heading && (
+          <li className={`${baseClass}__row ${baseClass}__header`}>
+            {heading}
+          </li>
+        )}
+        {items.map((_item) => {
+          // If an item has been marked as changed, use the changed version
+          // of the item rather than the one from the page fetch.  This allows
+          // us to render an item correctly even after we've navigated away
+          // from its page and then back again.
+          const item = dirtyItems[_item[idKey]] ?? _item;
+
+          const rowClasses = classnames(`${baseClass}__row`, {
+            [`${baseClass}__row--disabled`]: item.disabled,
+          });
+          return (
+            // eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions
+            <li
+              className={rowClasses}
+              key={item[idKey]}
+              onClick={() => {
+                if (item.disabled) return;
```

Comment:
`item.disabled` -> `isItemDisabled(item)` in these two places.

Alternatively set that in a var and use it in all 3 places.